### PR TITLE
refactor:  Extract and encapsulate filtering query management and filtering state logic

### LINKED ISSUE #624: Refactor submission list filtering: URL ↔ state ↔ Endatix API
### Goal

Consolidate submission grid filtering into **testable, pure** code paths: one canonical request shape aligned with `ListSubmissionsRequest`, shared **Hub URL** serialization/parsing, and a single **Endatix list** query encoder—so the submissions page, client navigation, and `lib/endatix-api` stay in sync as filters and related APIs evolve (e.g. export).

### Implementation Notes

- **Correctness**: Fix calendar-day → UTC ISO conversion used for date filters in `hub/lib/endatix-api/submissions/submissions.ts` (use `Date.UTC`, consistent with server-side date parsing on the submissions page). Add focused unit tests.
- **Feature slice** (per `hub/project-structure.md`): Add `hub/features/submissions/list-submission-query/` with parse/serialize helpers, shared defaults (`DEFAULT_PAGE`, `DEFAULT_PAGE_SIZE`, page size rules), optional canonical-URL checks, and `__tests__/` (round-trip URL ↔ `ListSubmissionsRequest`). Wire `app/(main)/forms/[formId]/submissions/page.tsx` and `features/submissions/ui/submissions-with-filters.tsx` to use this slice—remove duplicated `URLSearchParams` logic and magic `10` vs server defaults.
- **API client**: Extract pure `appendSubmissionListFilters` (or equivalent) next to submission list types; keep `Submissions.list` as a thin wrapper. Co-locate tests.
- **UI cleanup**: Remove dead empty-state branch in `features/submissions/ui/table/data-table.tsx`; prefer `??` for external sorting handler; optionally use `NoMatchingSubmissionsEmptyState` when the table is empty but total filtered count indicates “filters only” empty. Small nits: stable empty objects in `draggable-column-header.tsx`, `tableKey` helper, date-filter JSDoc, test assertion fixes.
- **Future (optional OSS coordination)**: Hub currently may issue a second unfiltered list probe for empty-state UX. A later API enhancement (e.g. `hasAnySubmissionsIgnoringFilters` on the list response) could remove that; track separately if needed—this issue focuses on Hub refactor + client contract alignment.

Related context: submission grid work landed around PR #623; this issue captures the follow-up cleanup and architecture alignment.

### PR DESCRIPTION
**Pull request by PeterMarinov**
# refactor: Extract and encapsulate filtering query management and filtering state logic

## Description
- Define common logic for filtering state detection and manageming (SUBMISSION_LIST_FILTER_QUERY_PARAM, SUBMISSION_LIST_FILTER_FIELD_NAMES)
- Extract common logic for handling UTC day-boundary cases + tests
- Adds centralized Hub route search param keys (SUBMISSION_LIST_URL_SEARCH_PARAM_KEYS), aligned parse/serialize on them, validated client
- DRY for URL state via the same parsers/allowlists as the server, simplified the submissions table cache key string, aligned the submissions page default page with slice constants, and expanded Vitest (path, canonical URL, client bridge). 
- Employes the Empty component for consistent UI when there are no results
- Submissions table hydration — Fixed @dnd-kit SSR/client `aria-describedby` mismatch by giving DndContext a stable id from React `useId()`

## Related Issues
- closes https://github.com/endatix/endatix-hub/issues/624

## Type of Change
Check all that apply.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Dependency update
- [x] Refactoring (non-breaking change which improves code quality)
- [ ] Other (please describe)

## Checklist
- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

> [!NOTE]
> - [x] I confirm that my Pull Request follows all the checklist requirements above

## Screenshots
If applicable, add screenshots to help explain your changes.

## Release Notes

* **New Features**
  * Added ability to clear filters when no submissions match current filter criteria.

* **Improvements**
  * Updated empty state messaging for enhanced clarity when no submissions are available.
  * Refined empty state styling with improved visual presentation.
  * Enhanced pagination and filter URL handling for more reliable browsing experience.

* **Tests**
  * Added comprehensive test coverage for submissions list pagination and filtering behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/endatix/endatix-hub/pull/625)

### CODE REVIEW

**Review by PeterMarinov — COMMENTED** [review]
Reviewed

File: lib/endatix-api/submissions/submissions.ts [hunk 1]
```diff
@@ -182,43 +183,7 @@ export class Submissions {
     const params = new URLSearchParams();
     params.set("page", String(request.page ?? 1));
     params.set("pageSize", String(request.pageSize ?? 10));
```
[the remark this answers is not in the document]
Comment (PeterMarinov) [reply lib/endatix-api/submissions/submissions.ts]: Yes, not it uses the default values set in the submission list constants

File: lib/endatix-api/submissions/submission-list-query-params.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    params.append(
+      SUBMISSION_LIST_FILTER_QUERY_PARAM,
+      `${SUBMISSION_LIST_FILTER_FIELD_NAMES.completedAt}>:${utcCalendarDayStartIso(request.completedAtFrom)}`,
+    );
+  }
+  if (request.completedAtTo) {
+    params.append(
+      SUBMISSION_LIST_FILTER_QUERY_PARAM,
+      `${SUBMISSION_LIST_FILTER_FIELD_NAMES.completedAt}<${utcCalendarNextDayStartIso(request.completedAtTo)}`,
+    );
```
[the remark this answers is not in the document]
Comment (PeterMarinov) [reply lib/endatix-api/submissions/submission-list-query-params.ts]: Yes, correct findings. Added regex testing for validaiton

File: features/submissions/ui/table/column-order-context.tsx [hunk 3]
```diff
@@ -125,13 +127,18 @@ export function ColumnOrderProvider<TData extends Submission = Submission>({
...
+      resetToDefault,
+      hasCustomOrder,
+    ],
   );
 
   return (
-    <ColumnOrderContext value={contextValue}>
-      {children}
-    </ColumnOrderContext>
+    <ColumnOrderContext value={contextValue}>{children}</ColumnOrderContext>
```
[the remark this answers is not in the document]
Comment (PeterMarinov) [reply features/submissions/ui/table/column-order-context.tsx:141]: Starting in React 19, you can render <SomeContext> as a provider. In older versions of React, use <SomeContext.Provider>.

I think, this is OK - https://react.dev/reference/react/createContext
